lib/swoosh/adapters/mandrill.ex

```diff
@@ -94,11 +94,22 @@ defmodule Swoosh.Adapters.Mandrill do
 
   defp prepare_attachments(body, %{attachments: []}), do: body
   defp prepare_attachments(body, %{attachments: attachments}) do
-    Map.put(body, "attachments", Enum.map(attachments, &%{
-      "name" => &1.filename,
-      "type" => &1.content_type,
-      "content" => &1.path |> File.read! |> Base.encode64
-    }))
+    normal_attachments = Enum.filter(attachments, fn(attachment) -> attachment.type == :attachment end)
+    inline_attachments = Enum.filter(attachments, fn(attachment) -> attachment.type == :inline end)
+    
+    normal_attachments = Enum.map(normal_attachments, fn %Swoosh.Attachment{content_type: type, path: path, filename: filename} ->
+      content = path |> File.read! |> Base.encode64
+      %{type: type, name: filename, content: content}
+    end)
+
+    inline_attachments = Enum.map(inline_attachments, fn %Swoosh.Attachment{content_type: type, path: path, filename: filename} ->
+      content = path |> File.read! |> Base.encode64
+      %{type: type, name: "#{filename}", content: content}
```

Comment:
- I'd extract this common part into a separate function
- `"#{filename}"` is just `filename`?

Reply:
This has also been updated. Thanks.